libs/dotnet-sdk/FeatureBoardService.cs

```diff
@@ -0,0 +1,36 @@
+using FeatureBoard.DotnetSdk.Models;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace FeatureBoard.DotnetSdk;
+
+internal class FeatureBoardService : IFeatureBoardService
+{
+  private readonly IFeatureBoardHttpClient _httpClient;
+  private readonly ILogger _logger;
+
+  private static readonly SemaphoreSlim _semaphore = new(1, 1);
+
+  public FeatureBoardService(IFeatureBoardHttpClient httpClient, ILogger<FeatureBoardService> logger)
+  {
+    _httpClient = httpClient;
+    _logger = logger;
+  }
+
+  public virtual async Task<bool?> RefreshFeatureConfiguration(CancellationToken cancellationToken)
```

Comment:
Nice, for context this particular update strategy I designed for Lambda which doesn't allow you to run background processes because the runtime gets paused between requests and each 'instance' can only run a single request.

BUT it's a better idea to make this strategy safe if someone uses it outside of it's planned environment, so this is a great change.

Reply:
I don't follow what you're talking about here regarding lambda? This class is basically a relocation and slight reworking of the logic which was previously in the `OnRequestUpdateStrategryMiddleware` which was being injected in to the ASP.NET MVC request pipeline...

The "reworking" was to change so it no longer blocks all requests for the duration of the request/response from Featureboard API...